DigitalLearningSolutions.Data.Tests/Services/RegistrationServiceTests.cs

```diff
@@ -1126,6 +1164,62 @@ public void CreateDelegateAccountForExistingUser_with_unapproved_IP_registers_de
             approved.Should().BeFalse();
         }
 
+        [Test]
+        public void CreateDelegateAccountForExistingUser_adds_new_delegate_to_groups()
+        {
+            // Given
+            const int userId = 2;
+            const int delegateId = 2;
+            var model = RegistrationModelTestHelper.GetDefaultInternalDelegateRegistrationModel();
+            var delegateEntity = UserTestHelper.GetDefaultDelegateEntity(
+                7,
+                answer1: "a",
+                answer2: "b",
+                answer3: "c",
+                answer4: "d",
+                answer5: "e",
+                answer6: null
+            );
+            var expectedAnswers = delegateEntity.GetRegistrationFieldAnswers();
+            A.CallTo(
+                () => registrationDataService.RegisterDelegateAccountAndCentreDetailForExistingUser(
+                    A<DelegateRegistrationModel>._,

```

Comment:
It would be better if these were the actual `delegateRegistrationModel` and `userId` being passed in